jaas/modules/src/main/java/org/apache/karaf/jaas/modules/ldap/LDAPBackingEngine.java

```diff
@@ -0,0 +1,210 @@
+/*
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *  under the License.
+ */
+package org.apache.karaf.jaas.modules.ldap;
+
+import org.apache.karaf.jaas.boot.principal.GroupPrincipal;
+import org.apache.karaf.jaas.boot.principal.RolePrincipal;
+import org.apache.karaf.jaas.boot.principal.UserPrincipal;
+import org.apache.karaf.jaas.modules.BackingEngine;
+import org.apache.karaf.jaas.modules.ldap.LDAPCache;
+import org.apache.karaf.jaas.modules.ldap.LDAPOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Karaf JAAS backing engine to support basic list funcitonality
+ * for the LDAP login module.  Modification is not supported
+ * at this time
+ */
+public class LDAPBackingEngine implements BackingEngine {
+
+    LDAPCache cache;
+    LDAPOptions options;
+
+    private static Logger LOGGER = LoggerFactory.getLogger(LDAPBackingEngine.class);
+
+    public LDAPBackingEngine(Map<String, ?> options) {
+        this.options = new LDAPOptions(options);
+        cache = LDAPCache.getCache(this.options);
+    }
+
+    @Override
+    public void addUser(String username, String password) {
+        throw new RuntimeException("Adding a user is not supporting in LDAP");
+    }
+
+    @Override
+    public void deleteUser(String username) {
+        throw new RuntimeException("Deleting a user is not supporting in LDAP");
```

Comment:
What about `UnsupportedOperationException` instead of a generic `RuntimeException` ?